test/DynamoCoreWpfTests/DynamoTestUIBase.cs

```diff
@@ -24,6 +24,82 @@
 
 namespace DynamoCoreWpfTests
 {
+    internal class TestDiagnostics
+    {
+        internal int DispatcherOpsCounter = 0;
+        // Use this flag to skip trying to execute all the dispatched operations during the test lifetime.
+        // This flag should only be used very sparingly
+        internal bool SkipDispatcherFlush = false;
+
+        private void Hooks_OperationPosted(object sender, DispatcherHookEventArgs e)
+        {
+            e.Operation.Task.ContinueWith((t) => {
+                Interlocked.Decrement(ref DispatcherOpsCounter);
+            }, TaskScheduler.Default);
+            Interlocked.Increment(ref DispatcherOpsCounter);
+        }
+
+        private void PrettyPrint(object obj)
+        {
+            Type type = obj.GetType();
+            PropertyInfo[] properties = type.GetProperties();
+
+            Console.WriteLine("{");
+            foreach (var property in properties)
+            {
+                try
+                {
+                    Console.WriteLine($"  {property.Name}: {property.GetValue(obj)}");
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine($"  {property.Name}: {e.Message}");
+                }
+            }
+            Console.WriteLine("}");
+        }
+
+        internal void SetupStartupDiagnostics()
+        {
+            System.Console.WriteLine($"PID {Process.GetCurrentProcess().Id} Start test: {TestContext.CurrentContext.Test.Name}");
+            TestUtilities.WebView2Tag = TestContext.CurrentContext.Test.Name;
+
+            SynchronizationContext.SetSynchronizationContext(new DispatcherSynchronizationContext());
```

Comment:
Is this because the tests don't run in the UI thread by default?

Reply:
This code was simply moved to a new utility class, but the reasons for it (when it was added way back) was that tests were running on non UI threads